scilpy/utils/spatial_ops.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+
+def parallel_transport_streamline(streamline, nb_streamlines, radius, rng=None):
+    """ Generate new streamlines by parallel transport of the input
+    streamline. See [0] and [1] for more details.
+
+    [0]: Hanson, A.J., & Ma, H. (1995). Parallel Transport Approach to 
+        Curve Framing. # noqa E501
+    [1]: TD Essentials: Parallel Transport.
+        https://www.youtube.com/watch?v=5LedteSEgOE
+
+    Parameters
+    ----------
+    streamline: ndarray (N, 3)
+        The streamline to transport.
+    nb_streamlines: int
+        The number of streamlines to generate.
+
+    Returns
+    -------
+    new_streamlines: list of ndarray (N, 3)
+        The generated streamlines.
+    """
+
+    def r(vec, theta):
+        """ Rotation matrix around a 3D vector by an angle theta.
+        From https://stackoverflow.com/questions/6802577/rotation-of-3d-vector
+
+        TODO?: Put this somewhere else.
+
+        Parameters
+        ----------
+        vec: ndarray (3,)
+            The vector to rotate around.
+        theta: float
+            The angle of rotation in radians.
+
+        Returns
+        -------
+        rot: ndarray (3, 3)
+            The rotation matrix.
+        """
+
+        vec = vec / np.linalg.norm(vec)
+        x, y, z = vec
+        c, s = np.cos(theta), np.sin(theta)
+        return np.array([[c + x**2 * (1 - c),
+                          x * y * (1 - c) - z * s,
+                          x * z * (1 - c) + y * s],
+                         [y * x * (1 - c) + z * s,
+                          c + y**2 * (1 - c),
+                          y * z * (1 - c) - x * s],
+                         [z * x * (1 - c) - y * s,
+                          z * y * (1 - c) + x * s,
+                          c + z**2 * (1 - c)]])
+    if rng is None:
+        rng = np.random.default_rng(0)
+
+    # Compute the tangent at each point of the streamline
+    T = np.gradient(streamline, axis=0)
+    # Normalize the tangents
+    T = T / np.linalg.norm(T, axis=1)[:, None]
+
+    # Placeholder for the normal vector at each point
+    V = np.zeros_like(T)
+    # Set the normal vector at the first point to be [0, 1, 0]
+    # (arbitrary choice)
```

Comment:
The first normal vector is no longer the Y base but a transposed (? sort-of ?) version of the first direction vector.